base/src/main/java/org/aya/concrete/desugar/Desugarer.java

```diff
@@ -126,10 +129,45 @@ private int levelVar(@NotNull Expr expr) throws DesugarInterruption {
   public static class DesugarInterruption extends Exception {
   }
 
+  @Override
+  public @NotNull Pattern visitPattern(@NotNull Pattern pattern, Unit pp) {
+    return switch (pattern) {
+      case Pattern.List list -> {
+        if (list.nilName() instanceof Pattern.Ctor nilCtor) {
+          if (list.consName() instanceof Pattern.Ctor consCtor) {
```

Comment:
We can use guard patterns like
```java
case Pattern.List list when list.nilName() instanceof ... && list.consName() instanceof ... -> do something;
case Pattern.List ignored -> report error;
```

Reply:
I think we can skip these two checks and just proceed with `list.nilName()` and `list.consName()`. They are all "valid" but may contain type errors (the typechecker is responsible for it). 
We did the same to idiom brackets and do notation --- resolves to whatever in scope.

Reply:
Okay